operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/KubernetesDependentResourceController.java

```diff
@@ -46,45 +43,16 @@ public KubernetesDependentResourceController(DependentResource<R, P> delegate,
             configuration.getDependentResourceClass());
   }
 
-  @SuppressWarnings("unchecked")
-  @Override
-  protected Persister<R, P> initPersister(DependentResource<R, P> delegate) {
-    return (delegate instanceof Persister) ? (Persister<R, P>) delegate : this;
-  }
-
-  @Override
-  public String descriptionFor(R resource) {
-    return String.format("'%s' %s dependent in namespace %s", resource.getMetadata().getName(),
-        resource.getFullResourceName(),
-        resource.getMetadata().getNamespace());
-  }
-
-  @Override
-  public EventSource initEventSource(EventSourceContext<P> context) {
-    this.client = context.getClient();
-    informer = new InformerEventSource<>(configuration, context);
-    return informer;
-  }
-
   @Override
-  public void createOrReplace(R dependentResource, Context context) {
-    client.resource(dependentResource).createOrReplace();
+  public Optional<EventSource> eventSource(EventSourceContext<P> context) {
```

Comment:
Why are we returning an optional? In what situation(s) would one not want to create an event source?

Reply:
For example, when the controller is just periodically checking some external resource(s). I think we should have this possibility to have it without event source, although most of the cases it will have.

Reply:
But you would still need an event source in that case, though… so I guess I don't understand how that's supposed to work.

Reply:
Still not sure how that would work…

Reply:
So basically imagine that there will be just an operator that reconciles every 5 mins, all the resources. All external, so no caches locally. This can pretty much work without an event source.

Reply:
Yes but that's not the most common case and in the mean time, the most common cases is polluted by the `Optional` handling… Let's see if we can address this in a subsequent PR.